### ISSUE TITLE
editMessageMedia:  Can't parse media JSON object

### ISSUE THREAD

**Issue by nesc1** [reo7sp/tgbot-cpp#153]
Hi all,

I'm trying to edit a photo with a caption with my bot and I'm using editMessageMedia(..) function.

What I'm supplying is the following:

```cpp
  auto media         = std::make_shared<TgBot::InputMedia>();
  media->type      = TgBot::InputMedia::TYPE::PHOTO;
  media->media     = "somefileid";
  media->parseMode = "MarkdownV2";
  media->caption = "sometext...";

 getApi().editMessageMedia(media, chatId, messageId);
```

The somefile id is for example this: `AgACAgQAAxkBAAILw1-LeF77Xie5gvKtJt9Y0kyfgZQrAAK0tDEbt5BZUAiTyiFJ_EjitAnfIl0AAwEAAwIAA20AA5TZBgABGwQ`

But i'm always getting the following error:

`Bad Request: can't parse InputMedia: Can't parse media JSON object`

Can someone tell me what I'm doing wrong?

Best regards,
Nuno

**Comment by nesc1** [comment]
Hi,
 
the problem in my case is that the `appendToJson` function when used with a `const char*` uses the template version of the `appendToJson` function, and this way does not set the type json value as a string, it puts something like:
```json
{"type":photo,
```
given the referenced error, so probably you will need to add a specialization for the `appendToJson` method, something like:
```cpp
    inline void appendToJson(std::string& json, const std::string& varName, const char* value) const {
        if(value!=nullptr){
            std::string strvalue(value);
            appendToJson(json,varName,strvalue);
        }
    }
```

**Comment by nesc1** [comment]
This is resolved with: #154
